### PR TITLE
added success notification  after saving data, used  Benefit.json fil…

### PR DESCRIPTION
**Pull request by DwinaTech**
…e as real data, edited area duplication , edit editOrganisation component to edit the data of the org the should edit it, also edited show detail component to show the real data of the related organisation.

### CODE REVIEW

File: frontend/src/Components/Organisation/EditOrganisation.js [hunk 1]
```diff
@@ -49,23 +85,30 @@ export default class EditOrganisation extends React.Component {
...
+  };
+
   render() {
     return (
       <div>
-        <button className="edit-button" onClick={this.handleClickOpen}>
+        <button className="edit-button" onClick={this.props.getData}>
           Edit
         </button>
+        <NotificationSystem ref="savedChangesSuccessfully" className="success-notification" />
```
Comment (kkarimi) [inline frontend/src/Components/Organisation/EditOrganisation.js]: Right now this is shown in the background, which is not good, should be in another component or atleast modal infront of it must be hidden

File: frontend/src/Components/Organisation/EditOrganisation.js [hunk 2]
```diff
@@ -75,15 +118,15 @@ export default class EditOrganisation extends React.Component {
...
-              name={this.state.name}
-              area={this.state.area}
-              borough={this.state.borough}
-              advices={this.state.advices}
-              process={this.state.process}
-              day={this.state.day}
-              telephone={this.state.telephone}
-              email={this.state.email}
-              website={this.state.website}
+              name={this.state.Organisation}
```
Comment (kkarimi) [inline frontend/src/Components/Organisation/EditOrganisation.js]: 4th time I see this list of things in the same file.. :( not very DRY

File: frontend/src/Components/Organisation/OrganisationForm.js [hunk 3]
```diff
@@ -8,17 +8,17 @@ import { FormControl } from 'material-ui/Form';
 import orgData from '../../Data/json/Debt.json'
 
 const boroughs = orgData.data;
-const areas = orgData.data;
 const processData = orgData.data;
-const days = ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday',]
+const days = ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday', 'Mon-Fri'];
+const areas = ['South', 'North', 'West', 'East', 'Central', 'All']
```
Comment (kkarimi) [inline frontend/src/Components/Organisation/OrganisationForm.js]: There are ways to de-duplicate an Array, I guess you did not like those!

File: frontend/src/Components/Organisation/SingleOrganisation.js [hunk 4]
```diff
@@ -11,6 +11,21 @@ import './single-org.css';
 export default class SingleOrganisation extends Component {
   state = {
     open: false,
+    editIdx: -1,
```
Comment (kkarimi) [inline frontend/src/Components/Organisation/SingleOrganisation.js]: Idx, idex and index - it's a nice variety but can we just use the same for consistency? (use index)

File: frontend/src/Components/Organisation/SingleOrganisation.js [hunk 5]
```diff
@@ -21,75 +36,81 @@ export default class SingleOrganisation extends Component {
...
-          <EditOrganisation />
-          <DialogTitle
-            className="single-oganisation-title"
-            id="alert-dialog-title"
+    const data = this.props.editOrgData;
+    const index = 1;
+    const { editIdx } = this.state;
+    const currentlyEditing = editIdx === index;
+
+    return currentlyEditing ? <EditOrganisation editOrgData={data} stopEditing={this.stopEditing} handleClichOpen={true} /> :
```
Comment (kkarimi) [inline frontend/src/Components/Organisation/SingleOrganisation.js]: clich => click, also... why is a boolean but with a function name? and if it is boolean, the correct way is per linter suggestion not to pass ={true} but just pass the name (a good name though, not handleCichOpen)

File: frontend/src/Components/Organisation/SingleOrganisation.js [hunk 6]
```diff
@@ -21,75 +36,81 @@ export default class SingleOrganisation extends Component {
...
-            </button>
-          </DialogActions>
-          <EditOrganisation />
-          <DialogTitle
-            className="single-oganisation-title"
-            id="alert-dialog-title"
+    const data = this.props.editOrgData;
+    const index = 1;
+    const { editIdx } = this.state;
+    const currentlyEditing = editIdx === index;
```
Comment (kkarimi) [inline frontend/src/Components/Organisation/SingleOrganisation.js]: Why is there an enforcing of one? a comment to explain would be good

File: frontend/src/Components/Organisation/index.js [hunk 7]
```diff
@@ -1,48 +1,74 @@
...
 import Paper from 'material-ui/Paper';
 import Grid from 'material-ui/Grid';
 import EditOrganisation from './EditOrganisation';
 import SingleOrganisation from './SingleOrganisation';
 import Search from './Search';
-import orgData from './org_data';
 import TopNav from '../TopNav';
 import helpers from '../../helpers';
-
+import orgData from '../../Data/json/Benefits.json'
```
Comment (kkarimi) [inline frontend/src/Components/Organisation/index.js]: Why this particular file? is it too difficult to show different data files for different service types?

File: frontend/src/Data/json/Benefits.json [hunk 8]
```diff
@@ -10,7 +10,7 @@
       "Borough": "Haringey",
       "Process": [],
       "Tel": [],
-      "Day": ["Monday", "Wednesday"],
+      "Day": ["Mon-Fri"],
```
Comment (kkarimi) [inline frontend/src/Data/json/Benefits.json]: Do not modify this file, it's a waste of time. You can use Javascript to clean up these fields as they will be dirty in the DB too

File: frontend/src/Components/Organisation/SingleOrganisation.js (hunk 5 above)
Comment (DwinaTech) [reply frontend/src/Components/Organisation/SingleOrganisation.js]: fixed

File: frontend/src/Components/Organisation/EditOrganisation.js [hunk 9]
```diff
@@ -1,45 +1,81 @@
...
-    telephone: "028 297 4111",
-    email: "test@test.co.uk",
-    website: "http://www.haringeymsc.org",
-    category: [],
+    notificationSystem: null,
+    open: this.props.handleClichOpen,
+    Organisation: "",
+    Area: "",
+    Borough: "",
+    Services: "",
```
[the remark this answers is not in the document]
Comment (DwinaTech) [reply frontend/src/Components/Organisation/EditOrganisation.js]: fixed

File: frontend/src/Components/Organisation/EditOrganisation.js [hunk 10]
```diff
@@ -1,45 +1,81 @@
...
-    name: "Haringey Migrant Service",
-    advices: "- Help accessing NHS",
-    process: "Call in advance (appt only)",
-    day: ["Monday"],
-    telephone: "028 297 4111",
-    email: "test@test.co.uk",
-    website: "http://www.haringeymsc.org",
-    category: [],
+    notificationSystem: null,
+    open: this.props.handleClichOpen,
```
[the remark this answers is not in the document]
Comment (DwinaTech) [reply frontend/src/Components/Organisation/EditOrganisation.js]: fixed

File: frontend/.eslintrc.js [hunk 11]
```diff
@@ -23,6 +23,8 @@ module.exports = {
     'linebreak-style': 'off',
     'consistent-return': 'off',
     'react/prop-types': 'off',
+    'react/no-string-refs': 'off',
+    'react/jsx-boolean-value': 'off'
```
[the remark this answers is not in the document]
Comment (DwinaTech) [reply frontend/.eslintrc.js]: removed

**Pull request merged**